Source/ASCollectionView.mm

```diff
@@ -1857,7 +1857,7 @@ - (void)_checkForBatchFetching
 
 - (void)_beginBatchFetchingIfNeededWithContentOffset:(CGPoint)contentOffset velocity:(CGPoint)velocity
 {
-  if (ASDisplayShouldFetchBatchForScrollView(self, self.scrollDirection, self.scrollableDirections, contentOffset, velocity)) {
+  if (ASDisplayShouldFetchBatchForScrollView(self, self.scrollDirection, self.scrollableDirections, contentOffset, velocity, self.collectionViewLayout.flipsHorizontallyInOppositeLayoutDirection)) {
```

Comment:
As @wsdwsd0829 pointed out to me, UICollectionView actually asks for this property in its initializer and caches it. Perhaps we should do the same for ASCollectionView? Not a blocker for this diff though.

 
![image(1)](https://user-images.githubusercontent.com/587874/116910711-d5ad2500-abfa-11eb-90a0-0c1e101b7682.png)
<img width="1919" alt="Screen Shot 2021-05-03 at 10 25 00 AM" src="https://user-images.githubusercontent.com/587874/116910715-d80f7f00-abfa-11eb-90e1-d9766a7214c2.png">

Reply:
Nvm, it's actually cached by UICollectionView**Layout** in its initializer and so asking for `flipsHorizontallyInOppositeLayoutDirection` on the layout object here should be fast.

Reply:
Do I need to worry about calling `self.collectionViewLayout. flipsHorizontallyInOppositeLayoutDirection` on a potentially off-main thread? Since it is cached and presumably not called again, do I not need to worry about it?

Reply:
I think the risk is very small. I checked and `_beginBatchFetchingIfNeededWithContentOffset` is called by multiple methods but all of them are called on the main thread. You can add a main thread assertion there just so we can be notified in the slim chance that's not the case.

Reply:
Worth noting @nguyenhuy and @rcancro `self.collectionViewLayout.flipsHorizontallyInOppositeLayoutDirection` is not availiable on iOS 10